### DISCUSSION TITLE
XDrip & Android 7 - Letsencrypt

### DISCUSSION THREAD

**Discussion by giuseppe-Marfisi** [NightscoutFoundation/xDrip#3559] [Bug reports]
I haven't been able to use xdrip on my **Android 7** tablet for some time. The following sentence appears: _NS entries download failed: javax.net.ssl.SSLHandshakeException: Java.security.cert.CertPathValidatorException: Trust anchor for certification path not found_.  
it seems that the problem comes from the lack of the certificate. _Letsencrypt_.
My tablet is a Huawei mediapad T3, with Android 7. The problem occurs with different versions of xdrip.
Can you tell me if there is a solution to have xdrip on my tablet?

  Thank you

**Comment by Navid200** [comment]
Letsencrypt has ended free support for Android 7.
https://community.letsencrypt.org/t/support-for-android-7-and-older-from-oct-2024/216446

I see two options:
1- Use a device that has Android 8 or newer.  There are many options.
2- Use a Nightscout server that does not rely on Letsecrypt.  There are many options.

**Reply by giuseppe-Marfisi** [reply]
But through Nightscout do I necessarily have to see the values ​​on the website without being able to adjust the alarms or is there some alternative mode or app to x-drip?  Thank you

**Reply by Navid200** [reply]
xDrip can upload to a Nightscout server that is visible in a browser.  
The limitation you are experiencing is not an xDrip limitation.  It is a limitation of your Nightscout server.

Do you not know that we are volunteers?  You are posting in xDrip forums asking for an alternative to xDrip?!
Is that your way of thanking the volunteers for working hard to develop and maintain xDrip?

If xDrip is not good enough for you, you can use an alternative. We have no financial incentive in convincing you to use xDrip.
